### ISSUE TITLE
Difference between plot_model in pycaret 3 vs 2 for classification tasks?

### ISSUE THREAD

**Issue by ngupta23** [pycaret/pycaret#3508]
### Discussed in https://github.com/pycaret/pycaret/discussions/3506

<div type='discussions-op-text'>

<sup>Originally posted by **sorenwacker** April 25, 2023</sup>
I am getting quite different results with plot_model functions between pycaret 3 compared to pycaret 2. The ROC cuve and confusion matrix seem not match the metrics output in version 3:

# pycaret 2
![image](https://user-images.githubusercontent.com/3391614/234368115-c5cae169-0fe1-4745-b7a1-2a6ba63b35a1.png)
![image](https://user-images.githubusercontent.com/3391614/234368161-99c649e9-9861-487f-bed1-2bf8ac063f7d.png)

# pycaret 3
![image](https://user-images.githubusercontent.com/3391614/234368218-4f2c9c9c-3bba-47f5-9ba0-c3113daa0c74.png)
![image](https://user-images.githubusercontent.com/3391614/234368238-94841c3f-168a-4bf9-bd69-6de049f34b5b.png)

# Code

## v2
```
data = df[categorical_features+numeric_features+[target]].reset_index(drop=True)

setup(data, 
      target=target, 
      categorical_features=categorical_features, 
      numeric_features=numeric_features,
      #log_experiment = True, 
      silent=True,
     )

model = create_model('lightgbm', fold=3)

plot_model(model, 'confusion_matrix')

plot_model(model, 'auc')
```

## v3

```
def run_experiment(categorical_features, numeric_features, target='target', **kwargs):
    data = df[categorical_features+numeric_features+[target]].reset_index(drop=True)
    
    setup(data, 
          target=target, 
          categorical_features=categorical_features, 
          numeric_features=numeric_features,
          log_experiment = True, 
          # fold_strategy='kfold',
    )

    model = create_model('lightgbm', fold=3)
    plot_model(model, 'confusion_matrix')
    plot_model(model, 'auc')
    plot_model(model, 'feature')

    predict_model(model)
    
    pipeline = get_config("pipeline")
    data_transformed = pipeline.transform(data)
    
    #shapley_analysis = ShapAnalysis(model, data_transformed)
    return model, data_transformed
```

# Question
What is different between plot_model functions from v2 and v3? It looks like in v3 the training data is used rather than the validation data. Is there something I have to change to reproduce the v2 behaviour?</div>

**Comment by sorenwacker** [comment]
#3507

**Comment by tvdboom** [comment]
Could you share the dataset you are using so I can reproduce the example?

**Comment by ngupta23** [comment]
@sorenwacker Has this issue been resolved with pycaret 3.0.2? It looks like the related issue (https://github.com/pycaret/pycaret/issues/3507) is resolved.

**Comment by sorenwacker** [comment]
Yes, is is resolved.
